tests/plugins_t/test_sqliteregistry.py

```diff
@@ -88,41 +93,81 @@ def setUp(self):
         config.Sites = [AttributeDict(name=self.test_site_name)]
         getattr(config, self.test_site_name).MachineTypes = [self.test_machine_type]
 
-    def test_add_machine_types(self):
-        registry = SqliteRegistry()
-        registry.add_site(self.test_site_name)
-        registry.add_machine_types(self.test_site_name, self.test_machine_type)
+        self.registry = SqliteRegistry()
 
+    def execute_db_query(self, sql_query):
         with sqlite3.connect(self.test_db) as connection:
             cursor = connection.cursor()
-            cursor.execute(
-                """SELECT MachineTypes.machine_type, Sites.site_name FROM MachineTypes
-                              JOIN Sites ON MachineTypes.site_id=Sites.site_id"""
+            cursor.execute(sql_query)
+
+            return cursor.fetchall()
+
+    def test_add_machine_types(self):
+        test_site_names = (self.test_site_name, self.other_test_site_name)
+
+        for site_name in test_site_names:
+            self.registry.add_site(site_name)
+            self.registry.add_machine_types(site_name, self.test_machine_type)
+
+        def check_db_content():
```

Comment:
Perhaps add a comment that the content needs to be checked several times. It's reasonably clear why this is a function but my finger was still itching to flatten it before reading on.

Reply:
Yes, that is true. I have added a comment to make it clear.